definitions/artifacts/cosmosdb-sql-authentication.json

```diff
@@ -0,0 +1,42 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "$md": {
+    "access": "public",
+    "name": "cosmosdb-sql-authentication"
+  },
+  "type": "object",
+  "title": "Cosmos DB SQL",
+  "description": "Cosmos DB SQL authentication and cloud-specific configuration.",
+  "additionalProperties": false,
+  "required": [
+    "data"
+  ],
+  "properties": {
+    "data": {
+      "type": "object",
+      "required": [
+        "infrastructure",
+        "authentication"
+      ],
+      "properties": {
+        "infrastructure": {
+          "title": "Cosmos DB SQL Infrastructure",
+          "description": "Cosmos DB SQL infrastructure configuration.",
+          "type": "object",
+          "oneOf": [
```

Comment:
This `oneOf` in the other database blocks was because there were multiple infrastructure patterns that could produce the artifact (different clouds, or kubernetes). It seems the only thing that will produce this artifact is cosmosdb, so we can probably drop the `oneOf` and just set it to the `ari` ref you have.

Reply:
Nit: On that note, since this isn't a generic multi-cloud definition, should we prefix it w/ `azure-`?

Reply:
@chrisghill would it look like this?

```

Reply:
Hmmm...I'm actually thinking no. The artifact itself is still an authentication artifact, which isn't necessarily cloud specific. It just happens to be that azure is the only one that offers it currently - and probably perpetually.

All of the other artifacts that are prefixed with the cloud are artifacts of the cloud service.  If this was an artifact of just cosmosdb (the service) then I would agree we prefix with `azure`.

Though we def need to make sure this artifact is named after this database engine. If the engine is commonly referred to as CosmosDB SQL, then that should be the name.

Reply:
> @chrisghill would it look like this?

No, that would technically produce an invalid schema since the ref'd type is a string.

Ideally we would use the type from here: https://github.com/massdriver-cloud/artifact-definitions/pull/84 but it needs to be merged first.

Reply:
Sorry, meant to erase the other junk:
```
      "properties": {
        "infrastructure": {
          "$ref": "../types/azure-resource-id.json"
        },
```

Reply:
No, you actually need that other junk. `infrastructure` is an object that (in this case) has one required field, which as called `ari` and is a string.

The way that is written would be turning `infrastructure` into the `ari` field itself, not an object containing the field.

Reply:
Ok I think I got it:
```
      "properties": {
        "infrastructure": {
          "title": "Cosmos DB SQL Infrastructure",
          "description": "Cosmos DB SQL infrastructure configuration.",
          "type": "object",
          "$ref": "../types/azure-infrastructure-ari.json"
        },
        "authentication": {
          "$ref": "../types/cosmosdb-sql-authentication.json"
        }
      }
```

Reply:
@chrisghill thanks for the ping on that one. Let me run the artifact dereference, then I'd need one more approval to merge:
https://github.com/massdriver-cloud/artifact-definitions/pull/84